src/main/java/org/apache/mesos/executor/CustomExecutor.java

```diff
@@ -7,18 +7,21 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.*;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.*;
 
 /**
  * An {@code Executor} implementation that supports execution of long-running tasks and supporting short-lived tasks.
  */
 public class CustomExecutor implements Executor {
     private static final Logger LOGGER = LoggerFactory.getLogger(
             CustomExecutor.class);
+    private static final int HEALTH_CHECK_THREAD_POOL_SIZE = 50;
```

Comment:
Yeah, I didn't want to run out of threads... what do you think 5?  There's really only 1 check running at a time normally.  Although it depends on the interval/timeout/delay settings.